cl/search/feeds.py

```diff
@@ -59,20 +61,36 @@ def items(self, obj):
                     )
                 else:
                     return []
-                main_params = search_utils.build_main_query(
-                    cd, highlight=False, facet=False
-                )
-                main_params.update(
-                    {
-                        "sort": f"{order_by} desc",
-                        "rows": "20",
-                        "start": "0",
-                        "caller": "SearchFeed",
+
+                if (
+                    waffle.flag_is_active(obj, "r-es-active")
+                    and cd["type"] == SEARCH_TYPES.RECAP
+                ):
+                    override_params = {
```

Comment:
We should refactor this if-statement and the if-elif statement starting in line 50 because we're always creating a `SolrInterface` even when the "r-es-active" flag is `True`. Here's my proposal:

```python
    if search_form.is_valid():
        cd = search_form.cleaned_data
        order_by = "dateFiled"
        es_search_query = None
        with Session() as session:
            # Using the pattern Matching we avoid creating the solr interface when r-es-active is True
            match cd["type"]:
                case SEARCH_TYPES.OPINION:
                    solr = ExtraSolrInterface(
                        settings.SOLR_OPINION_URL,
                        http_connection=session,
                        mode="r",
                    )
                case SEARCH_TYPES.RECAP if waffle.flag_is_active(obj, "r-es-active"):
                    es_search_query = ESRECAPDocument.search()
                case SEARCH_TYPES.RECAP:
                    solr = ExtraSolrInterface(
                        settings.SOLR_RECAP_URL,
                        http_connection=session,
                        mode="r",
                    )
                case _:
                    return []

            if es_search_query:
                override_params = {
                    "order_by": "entry_date_filed_feed desc",
                }
                cd.update(override_params)

                items = do_es_feed_query(
                    es_search_query,
                    cd,
                    rows=20,
                )
            else:
                main_params = search_utils.build_main_query(
                    cd, highlight=False, facet=False
                )
                main_params.update(
                    {
                        "sort": f"{order_by} desc",
                        "rows": "20",
                        "start": "0",
                        "caller": "SearchFeed",
                    }
                )
                # Eliminate items that lack the ordering field.
                main_params["fq"].append(f"{order_by}:[* TO *]")
                items = solr.query().add_extra(**main_params).execute()
            return items
```